From ef130c8884e85573523129f2d9eb2bf65e1b02ab Mon Sep 17 00:00:00 2001
From: Markus Volk <f_l_k@t-online.de>
Date: Sat, 28 Oct 2017 07:39:13 +0200
Subject: [PATCH] Makefile: add clean target

Signed-off-by: Markus Volk <f_l_k@t-online.de>
---
 Makefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Makefile b/Makefile
index 9897810..35bf169 100644
--- a/Makefile
+++ b/Makefile
@@ -7,3 +7,5 @@ sort-locale:
 	for language in $(locale); do \
 		LC_ALL=C sort -u -o $${language} $${language} ; \
 	done
+
+clean :
-- 
2.13.3

