From 3a69a859f063c726518e7a02503bf8e0b09404f5 Mon Sep 17 00:00:00 2001
From: Markus Volk <f_l_k@gmx.net>
Date: Tue, 22 Apr 2014 07:29:59 +0200
Subject: [PATCH 248/248] user_menu.cpp add missing lua ifdef

Signed-off-by: Markus Volk <f_l_k@gmx.net>
---
 src/gui/user_menue.cpp                 |  2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/src/gui/user_menue.cpp b/src/gui/user_menue.cpp
index 1149754..81e021a 100644
--- a/src/gui/user_menue.cpp
+++ b/src/gui/user_menue.cpp
@@ -464,7 +464,9 @@ bool CUserMenu::showUserMenu(int button)
 	if (games)			delete games;
 	if (tools)			delete tools;
 	if (scripts)			delete scripts;
+#if ENABLE_LUA
 	if (lua)			delete lua;
+#endif
 	if (menu)			delete menu;
 
 	InfoClock->enableInfoClock(true);
-- 
1.8.5.2

