From 8aea6c73eb723b887064a8fcb7abeef12ea0d3bc Mon Sep 17 00:00:00 2001
From: Markus Volk <f_l_k@gmx.net>
Date: Wed, 26 Nov 2014 08:25:34 +0100
Subject: [PATCH 841/841] luaclient.cpp: compile fix for gcc 4.9.1

Signed-off-by: Markus Volk <f_l_k@gmx.net>
---
 lib/luaclient/luaclient.cpp | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/lib/luaclient/luaclient.cpp b/lib/luaclient/luaclient.cpp
index 7344dca..f47bf86 100644
--- a/lib/luaclient/luaclient.cpp
+++ b/lib/luaclient/luaclient.cpp
@@ -79,21 +79,25 @@ int main(int argc, char** argv)
 
         if (!client.Send(data, size)) {
 		fun = "Send failed";
-		goto fail;
+		fprintf(stderr, "luaclient: %s.\n", fun);
+	exit(-1);
 	}
         if (!client.Recv((char *)&size, sizeof(size))) {
 		fun = "Recv (1) failed";
-		goto fail;
+		fprintf(stderr, "luaclient: %s.\n", fun);
+	exit(-1);
 	}
 	char result[size];
         if (!client.Recv(result, size)) {
 		fun = "Recv (2) failed";
-		goto fail;
+		fprintf(stderr, "luaclient: %s.\n", fun);
+	exit(-1);
 	}
 
 	if (result[size - 1]) {
 		fun = "unterminated result";
-		goto fail;
+		fprintf(stderr, "luaclient: %s.\n", fun);
+	exit(-1);
 	}
 	res = atoi(result);
 	resp = result + strlen(result) + 1;
@@ -103,8 +107,4 @@ int main(int argc, char** argv)
 	if (resp < result + size)
 		fprintf(stderr, "%s", resp);
 	exit(res);
-fail:
-	if (fun)
-		fprintf(stderr, "luaclient: %s.\n", fun);
-	exit(-1);
 }
-- 
2.0.0.rc2

