From 87a999d57ce8e5099cf19f4107e57701ea266754 Mon Sep 17 00:00:00 2001
From: Markus Volk <f_l_k@gmx.net>
Date: Mon, 23 Mar 2015 13:28:52 +0100
Subject: [PATCH 1093/1093] configure_fix

Signed-off-by: Markus Volk <f_l_k@gmx.net>
---
 configure.ac | 20 ++++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

diff --git a/configure.ac b/configure.ac
index 97b7455..2fe9231 100644
--- a/configure.ac
+++ b/configure.ac
@@ -54,7 +54,7 @@ if test "$TREMOR_STATIC" = "yes"; then
 fi
 
 if test "$TREMOR" = "yes"; then
-	TUXBOX_APPS_LIB_PKGCONFIG(VORBISIDEC,vorbisidec)
+	TUXBOX_APPS_LIB_PKGCONFIG_CHECK(VORBISIDEC,vorbisidec)
 	AC_DEFINE(USE_TREMOR,1,use libvorbisidec/tremor library)
 fi
 
@@ -65,20 +65,20 @@ fi
 
 TUXBOX_APPS_LIB_PKGCONFIG_CHECK(ID3TAG,libid3tag)
 if test x"$ID3TAG_EXISTS" != xyes; then
-	TUXBOX_APPS_LIB_PKGCONFIG(ID3TAG,id3tag)
+	TUXBOX_APPS_LIB_PKGCONFIG_CHECK(ID3TAG,id3tag)
 fi
 
 TUXBOX_APPS_LIB_PKGCONFIG_CHECK(MAD,libmad)
 if test x"$MAD_EXISTS" != xyes; then
-	TUXBOX_APPS_LIB_PKGCONFIG(MAD,mad)
+	TUXBOX_APPS_LIB_PKGCONFIG_CHECK(MAD,mad)
 fi
 
 TUXBOX_APPS_LIB_PKGCONFIG_CHECK(OGG,ogg)
 if test -z "$OGG_CFLAGS" ; then
-	TUXBOX_APPS_LIB_PKGCONFIG(OGG,ogg)
+	TUXBOX_APPS_LIB_PKGCONFIG_CHECK(OGG,ogg)
 fi
 else
-TUXBOX_APPS_LIB_PKGCONFIG(SWRESAMPLE,libswresample)
+TUXBOX_APPS_LIB_PKGCONFIG_CHECK(SWRESAMPLE,libswresample)
 fi
 
 if test "$BOXTYPE" = "tripledragon"; then
@@ -89,13 +89,13 @@ AM_CONDITIONAL(USE_TREMOR, test "$TREMOR" = "yes")
 
 # TUXBOX_APPS_LIB_PKGCONFIG(OPENSSL,openssl)
 TUXBOX_APPS_LIB_CONFIG(CURL,curl-config)
-TUXBOX_APPS_LIB_CONFIG(FREETYPE,freetype-config)
+TUXBOX_APPS_LIB_PKGCONFIG(FREETYPE,freetype2)
 
 TUXBOX_APPS_LIB_PKGCONFIG(PNG,libpng)
-TUXBOX_APPS_LIB_PKGCONFIG(AVFORMAT,libavformat)
-TUXBOX_APPS_LIB_PKGCONFIG(AVCODEC,libavcodec)
-TUXBOX_APPS_LIB_PKGCONFIG(AVUTIL,libavutil)
-TUXBOX_APPS_LIB_PKGCONFIG(BLURAY,libbluray)
+TUXBOX_APPS_LIB_PKGCONFIG_CHECK(AVFORMAT,libavformat)
+TUXBOX_APPS_LIB_PKGCONFIG_CHECK(AVCODEC,libavcodec)
+TUXBOX_APPS_LIB_PKGCONFIG_CHECK(AVUTIL,libavutil)
+TUXBOX_APPS_LIB_PKGCONFIG_CHECK(BLURAY,libbluray)
 TUXBOX_APPS_LIB_PKGCONFIG(SIGC,sigc++-2.0)
 
 #TUXBOX_APPS_LIB_PKGCONFIG(CONFIGFILE,tuxbox-configfile)
-- 
2.1.4

