From fc022e32525dd0351b7e5d3db53be464ae192909 Mon Sep 17 00:00:00 2001
From: Stefan Seyfried
Date: Sun, 1 Feb 2015 15:16:14 +0100
Subject: [PATCH] uncooloff.c: add #include <stdbool.h>

Signed-off-by: Markus Volk <f_l_k@gmx.net>
---
 src/uncooloff.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/uncooloff.c b/src/uncooloff.c
index 2c48e8c..9619348 100644
--- a/src/uncooloff.c
+++ b/src/uncooloff.c
@@ -14,6 +14,7 @@
 #include <time.h>
 #include <unistd.h>
 #include <sys/ioctl.h>
+#include <stdbool.h>
 
 #include <cs_frontpanel.h>
 
-- 
2.1.3

