From 735eb83e3f77f2bddfc425ec7d5daed28fdffd98 Mon Sep 17 00:00:00 2001
From: MarkusVolk <f_l_k@t-online.de>
Date: Mon, 27 Jul 2015 21:34:17 +0200
Subject: [PATCH 1362/1362] configure_fix

Signed-off-by: MarkusVolk <f_l_k@t-online.de>
---
 configure.ac | 24 ++++++++++++------------
 1 file changed, 12 insertions(+), 12 deletions(-)

diff --git a/configure.ac b/configure.ac
index 292cf82..768ea64 100644
--- a/configure.ac
+++ b/configure.ac
@@ -54,7 +54,7 @@ if test "$TREMOR_STATIC" = "yes"; then
 fi
 
 if test "$TREMOR" = "yes"; then
-	TUXBOX_APPS_LIB_PKGCONFIG(VORBISIDEC,vorbisidec)
+	TUXBOX_APPS_LIB_PKGCONFIG_CHECK(VORBISIDEC,vorbisidec)
 	AC_DEFINE(USE_TREMOR,1,use libvorbisidec/tremor library)
 fi
 
@@ -65,21 +65,21 @@ fi
 
 TUXBOX_APPS_LIB_PKGCONFIG_CHECK(ID3TAG,libid3tag)
 if test x"$ID3TAG_EXISTS" != xyes; then
-	TUXBOX_APPS_LIB_PKGCONFIG(ID3TAG,id3tag)
+	TUXBOX_APPS_LIB_PKGCONFIG_CHECK(ID3TAG,id3tag)
 fi
 
 TUXBOX_APPS_LIB_PKGCONFIG_CHECK(MAD,libmad)
 if test x"$MAD_EXISTS" != xyes; then
-	TUXBOX_APPS_LIB_PKGCONFIG(MAD,mad)
+	TUXBOX_APPS_LIB_PKGCONFIG_CHECK(MAD,mad)
 fi
 
 TUXBOX_APPS_LIB_PKGCONFIG_CHECK(OGG,ogg)
 if test -z "$OGG_CFLAGS" ; then
-	TUXBOX_APPS_LIB_PKGCONFIG(OGG,ogg)
+	TUXBOX_APPS_LIB_PKGCONFIG_CHECK(OGG,ogg)
 fi
-TUXBOX_APPS_LIB_PKGCONFIG(SWRESAMPLE,libswresample)
+TUXBOX_APPS_LIB_PKGCONFIG_CHECK(SWRESAMPLE,libswresample)
 else
-TUXBOX_APPS_LIB_PKGCONFIG(SWRESAMPLE,libswresample)
+TUXBOX_APPS_LIB_PKGCONFIG_CHECK(SWRESAMPLE,libswresample)
 fi
 
 if test "$BOXTYPE" = "tripledragon"; then
@@ -89,14 +89,14 @@ fi
 AM_CONDITIONAL(USE_TREMOR, test "$TREMOR" = "yes")
 
 # TUXBOX_APPS_LIB_PKGCONFIG(OPENSSL,openssl)
-TUXBOX_APPS_LIB_CONFIG(CURL,curl-config)
-TUXBOX_APPS_LIB_CONFIG(FREETYPE,freetype-config)
+TUXBOX_APPS_LIB_CONFIG_CHECK(CURL,curl-config)
+TUXBOX_APPS_LIB_PKGCONFIG(FREETYPE,freetype2)
 
 TUXBOX_APPS_LIB_PKGCONFIG(PNG,libpng)
-TUXBOX_APPS_LIB_PKGCONFIG(AVFORMAT,libavformat)
-TUXBOX_APPS_LIB_PKGCONFIG(AVCODEC,libavcodec)
-TUXBOX_APPS_LIB_PKGCONFIG(AVUTIL,libavutil)
-TUXBOX_APPS_LIB_PKGCONFIG(BLURAY,libbluray)
+TUXBOX_APPS_LIB_PKGCONFIG_CHECK(AVFORMAT,libavformat)
+TUXBOX_APPS_LIB_PKGCONFIG_CHECK(AVCODEC,libavcodec)
+TUXBOX_APPS_LIB_PKGCONFIG_CHECK(AVUTIL,libavutil)
+TUXBOX_APPS_LIB_PKGCONFIG_CHECK(BLURAY,libbluray)
 TUXBOX_APPS_LIB_PKGCONFIG(SIGC,sigc++-2.0)
 
 #TUXBOX_APPS_LIB_PKGCONFIG(CONFIGFILE,tuxbox-configfile)
-- 
2.1.4

