From 62b6600296978db77268ddfe57190d30dc27d0a9 Mon Sep 17 00:00:00 2001
From: flk <f_l_k@t-online.de>
Date: Tue, 17 Jan 2017 16:47:44 +0100
Subject: [PATCH] neutrino.cpp: adjust screen for fullhd

Signed-off-by: flk <f_l_k@t-online.de>
---
 src/neutrino.cpp | 8 +++++---------
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/neutrino.cpp b/src/neutrino.cpp
index 19593cb..eb3e048 100644
--- a/src/neutrino.cpp
+++ b/src/neutrino.cpp
@@ -308,10 +308,10 @@ static SNeutrinoSettings::usermenu_t usermenu_default[] = {
 #define DEFAULT_X_END_SD	1220
 #define DEFAULT_Y_END_SD	560
 
-#define DEFAULT_X_START_HD	40   //5
-#define DEFAULT_Y_START_HD	25   //5
-#define DEFAULT_X_END_HD	1235 //1275
-#define DEFAULT_Y_END_HD	690  //715
+#define DEFAULT_X_START_HD	45   //5
+#define DEFAULT_Y_START_HD	30   //5
+#define DEFAULT_X_END_HD	1875 //1275
+#define DEFAULT_Y_END_HD	1050  //715
 
 std::string ttx_font_file = "";
 
-- 
2.9.3

