From 710699c8431638d0e6e286a0365ab0aefb53d028 Mon Sep 17 00:00:00 2001
From: MarkusVolk <f_l_k@t-online.de>
Date: Mon, 28 Dec 2015 18:13:10 +0100
Subject: [PATCH 003/017] opkg 0.3.x uses opkg instead of opkg-cl as binary
 name

Signed-off-by: MarkusVolk <f_l_k@t-online.de>
---
 src/gui/opkg_manager.cpp | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/src/gui/opkg_manager.cpp b/src/gui/opkg_manager.cpp
index 1b92581..a4b4ea1 100644
--- a/src/gui/opkg_manager.cpp
+++ b/src/gui/opkg_manager.cpp
@@ -57,12 +57,7 @@
 #include <sys/wait.h>
 #include <fstream>
 
-#if 0
 #define OPKG_CL "opkg"
-#else
-#define OPKG_CL "opkg-cl"
-#endif
-
 #define OPKG_TMP_DIR "/tmp/.opkg"
 #define OPKG_TEST_DIR OPKG_TMP_DIR "/test"
 #define OPKG_CL_CONFIG_OPTIONS " -V2 --tmp-dir=/tmp --cache=" OPKG_TMP_DIR
-- 
2.1.4

