From 4d058e637ad54f85673221a846cf1b37744823d8 Mon Sep 17 00:00:00 2001
From: MarkusVolk <f_l_k@t-online.de>
Date: Tue, 5 Jan 2016 21:21:07 +0100
Subject: [PATCH 283/283] dont install unmaintained locale

Signed-off-by: MarkusVolk <f_l_k@t-online.de>
---
 data/locale/Makefile.am | 1 -
 1 file changed, 1 deletion(-)

diff --git a/data/locale/Makefile.am b/data/locale/Makefile.am
index 95e3a11..8698968 100644
--- a/data/locale/Makefile.am
+++ b/data/locale/Makefile.am
@@ -21,7 +21,6 @@ locale_unmaintained = \
 	unmaintained/svenska.locale
 
 install_DATA  = $(locale)
-install_DATA += $(locale_unmaintained)
 
 
 if MAINTAINER_MODE
-- 
2.1.4

