From e4b6d78190f0d453d25fcd1e14dd4c9ae455ad00 Mon Sep 17 00:00:00 2001
From: Markus Volk <f_l_k@gmx.net>
Date: Sat, 14 Feb 2015 15:06:33 +0100
Subject: [PATCH 1006/1006] remove unneeded mp3.jpg files

Signed-off-by: Markus Volk <f_l_k@gmx.net>
---
 data/icons/Makefile.am | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/data/icons/Makefile.am b/data/icons/Makefile.am
index e1166af..129bb15 100644
--- a/data/icons/Makefile.am
+++ b/data/icons/Makefile.am
@@ -163,16 +163,7 @@ install_DATA += \
 	mp_play.png \
 	mp_play_repeat_all.png \
 	mp_play_repeat_track.png \
-	mp3.jpg \
 	mp3.png \
-	mp3-0.jpg \
-	mp3-1.jpg \
-	mp3-2.jpg \
-	mp3-3.jpg \
-	mp3-4.jpg \
-	mp3-5.jpg \
-	mp3-6.jpg \
-	mp3-7.jpg \
 	multimedia.png \
 	mute.png \
 	mute_small.png \
-- 
2.1.3

