From 36cf243f8abca3d5124c56b69cd1949422a2dbf8 Mon Sep 17 00:00:00 2001
From: Markus Volk <f_l_k@gmx.net>
Date: Thu, 26 Mar 2015 10:16:20 +0100
Subject: [PATCH 1097/1097] set image version

Signed-off-by: Markus Volk <f_l_k@gmx.net>
---
 src/gui/imageinfo.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/gui/imageinfo.cpp b/src/gui/imageinfo.cpp
index c99fdd3..d18e968 100644
--- a/src/gui/imageinfo.cpp
+++ b/src/gui/imageinfo.cpp
@@ -263,7 +263,7 @@ void CImageInfo::InitInfoData()
 
 	image_info_t imagename 	= {LOCALE_IMAGEINFO_IMAGE,	config.getString("imagename", "Neutrino-HD")};
 	v_info.push_back(imagename);
-	image_info_t version	= {LOCALE_IMAGEINFO_VERSION,	version_string};
+	image_info_t version	= {LOCALE_IMAGEINFO_VERSION,	config.getString("version", "n/a")};
 	v_info.push_back(version);
 #ifdef VCS
 	image_info_t vcs	= {LOCALE_IMAGEINFO_VCS,	VCS};
-- 
2.1.4

