From 92d44e9472af4152f72a42752eee72078d4d35ed Mon Sep 17 00:00:00 2001
From: MarkusVolk <f_l_k@t-online.de>
Date: Sun, 8 Nov 2015 11:12:28 +0100
Subject: [PATCH 1449/1449] fix build with old libs (glibc)

Signed-off-by: MarkusVolk <f_l_k@t-online.de>
---
 src/gui/movieplayer.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/gui/movieplayer.cpp b/src/gui/movieplayer.cpp
index 6d2e6f2..58f1e7e 100644
--- a/src/gui/movieplayer.cpp
+++ b/src/gui/movieplayer.cpp
@@ -2196,7 +2196,7 @@ void CMoviePlayerGui::makeScreenShot(bool autoshot, bool forcover)
 		autoshot_done = true;
 }
 
-size_t CMoviePlayerGui::GetReadCount()
+/*size_t CMoviePlayerGui::GetReadCount()
 {
         uint64_t this_read = 0;
         this_read = playback->GetReadCount();
@@ -2208,4 +2208,4 @@ size_t CMoviePlayerGui::GetReadCount()
         last_read = this_read;
 //printf("GetReadCount: %lld\n", res);
         return (size_t) res;
-}
+}*/
 src/gui/streaminfo2.cpp | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/src/gui/streaminfo2.cpp b/src/gui/streaminfo2.cpp
index 6b75811..0ac6bc1 100644
--- a/src/gui/streaminfo2.cpp
+++ b/src/gui/streaminfo2.cpp
@@ -878,9 +878,9 @@ int CStreamInfo2::ts_setup ()
 #endif
 
 	short ret = -1;
-	if (mp) {
+/*	if (mp) {
 		mp->GetReadCount();
-	} else {
+	} else*/ {
 
 		unsigned short vpid, apid = 0;
 
@@ -932,10 +932,10 @@ int CStreamInfo2::update_rate ()
 	if(!dmx && !mp)
 		return 0;
 
-        if (mp) {
-                usleep(timeout * 1000);
-                b_len = mp->GetReadCount();
-        } else {
+/*	if (mp) {
+		usleep(timeout * 1000);
+		b_len = mp->GetReadCount();
+		} else*/ {
 		b_len = dmx->Read(dmxbuf, TS_BUF_SIZE, timeout);
 	}
 	//printf("ts: read %d\n", b_len);
-- 
2.1.4

