From 92d44e9472af4152f72a42752eee72078d4d35ed Mon Sep 17 00:00:00 2001
From: MarkusVolk <f_l_k@t-online.de>
Date: Sun, 8 Nov 2015 11:12:28 +0100
Subject: [PATCH 1449/1449] fix build with old libs (glibc)

Signed-off-by: MarkusVolk <f_l_k@t-online.de>
---
 src/gui/movieplayer.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/gui/movieplayer.cpp b/src/gui/movieplayer.cpp
index a254e4e..8ffe2b1 100644
--- a/src/gui/movieplayer.cpp
+++ b/src/gui/movieplayer.cpp
@@ -668,7 +668,7 @@ bool CMoviePlayerGui::StartWebtv(void)
 
 	playback->Open(is_file_player ? PLAYMODE_FILE : PLAYMODE_TS);
 
-	bool res = playback->Start((char *) file_name.c_str(), cookie_header);//url with cookies
+	bool res = 0;
 
 	playback->SetSpeed(1);
 	if (!res) {
@@ -2305,6 +2305,7 @@ void CMoviePlayerGui::selectSubtitle()
 	APIDSelector.exec(NULL, "");
 	delete selector;
 	printf("CMoviePlayerGui::selectSubtitle: selected %d (%x) current %x\n", select, (select >= 0) ? spids[select] : -1, currentspid);
+#if 0
 	if ((select >= 0) && (select < numsubs) && (currentspid != spids[select])) {
 		currentspid = spids[select];
 		/* external subtitles pid is 0x1FFF */
@@ -2317,6 +2318,7 @@ void CMoviePlayerGui::selectSubtitle()
 		playback->SelectSubtitles(currentspid, g_settings.subs_charset);
 		printf("[movieplayer] spid changed to %d\n", currentspid);
 	}
+#endif
 }
 
 extern "C" {
@@ -2545,7 +2547,7 @@ void CMoviePlayerGui::selectAutoLang()
 {
 	if (!numsubs)
 		playback->FindAllSubs(spids, sub_supported, &numsubs, slanguage);
-
+#if 0
 	if (ext_subs) {
 		for (unsigned count = 0; count < numsubs; count++) {
 			if (spids[count] == 0x1FFF) {
@@ -2554,6 +2556,7 @@ void CMoviePlayerGui::selectAutoLang()
 			}
 		}
 	}
+#endif
 	if (g_settings.auto_lang &&  (numpida > 1)) {
 		int pref_idx = -1;
 
@@ -2606,11 +2609,13 @@ void CMoviePlayerGui::selectAutoLang()
 				if (currentspid > 0)
 					break;
 			}
+#if 0
 			if (currentspid > 0) {
 				playback->SelectSubtitles(currentspid, g_settings.subs_charset);
 				printf("[movieplayer] spid changed to %d %s (%s)\n", currentspid, temp.c_str(), slang.c_str());
 				break;
 			}
+#endif
 		}
 	}
 }
@@ -2719,7 +2724,7 @@ void CMoviePlayerGui::makeScreenShot(bool autoshot, bool forcover)
 	if (autoshot)
 		autoshot_done = true;
 }
-
+#if 0
 size_t CMoviePlayerGui::GetReadCount()
 {
         uint64_t this_read = 0;
@@ -2733,7 +2738,7 @@ size_t CMoviePlayerGui::GetReadCount()
 //printf("GetReadCount: %lld\n", res);
         return (size_t) res;
 }
-
+#endif
 void CMoviePlayerGui::screensaver(bool on)
 {
 	if (on)
diff --git a/src/gui/streaminfo2.cpp b/src/gui/streaminfo2.cpp
index 6b75811..0ac6bc1 100644
--- a/src/gui/streaminfo2.cpp
+++ b/src/gui/streaminfo2.cpp
@@ -878,9 +878,9 @@ int CStreamInfo2::ts_setup ()
 #endif
 
 	short ret = -1;
-	if (mp) {
+/*	if (mp) {
 		mp->GetReadCount();
-	} else {
+	} else*/ {
 
 		unsigned short vpid, apid = 0;
 
@@ -932,10 +932,10 @@ int CStreamInfo2::update_rate ()
 	if(!dmx && !mp)
 		return 0;
 
-        if (mp) {
-                usleep(timeout * 1000);
-                b_len = mp->GetReadCount();
-        } else {
+/*	if (mp) {
+		usleep(timeout * 1000);
+		b_len = mp->GetReadCount();
+		} else*/ {
 		b_len = dmx->Read(dmxbuf, TS_BUF_SIZE, timeout);
 	}
 	//printf("ts: read %d\n", b_len);
-- 
2.1.4

