From 6bc7e4e6fc479fe9fb70915c08505aff8fbc1908 Mon Sep 17 00:00:00 2001
From: Stefan Seyfried <seife@tuxbox-git.slipkontur.de>
Date: Sun, 15 Nov 2015 11:45:12 +0100
Subject: [PATCH] acinclude: fix DVB_API_VERSION check for gcc5

---
 acinclude.m4 | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/acinclude.m4 b/acinclude.m4
index bcc7c03..2d55c53 100644
--- a/acinclude.m4
+++ b/acinclude.m4
@@ -232,13 +232,13 @@ AC_CHECK_HEADERS(linux/dvb/version.h,[
 #include <linux/dvb/version.h>
 version DVB_API_VERSION
 	]])])
-	DVB_API_VERSION=`(eval "$ac_cpp conftest.$ac_ext") 2>&AS_MESSAGE_LOG_FD | $EGREP "^version" | sed "s,version\ ,,"`
+	DVB_API_VERSION=`(eval "$ac_cpp -traditional-cpp conftest.$ac_ext") 2>&AS_MESSAGE_LOG_FD | $EGREP "^version" | sed "s,version\ ,,"`
 
 	AC_LANG_CONFTEST([AC_LANG_SOURCE([[
 #include <linux/dvb/version.h>
 version DVB_API_VERSION_MINOR
 	]])])
-	DVB_API_VERSION_MINOR=`(eval "$ac_cpp conftest.$ac_ext") 2>&AS_MESSAGE_LOG_FD | $EGREP "^version" | sed "s,version\ ,,"`
+	DVB_API_VERSION_MINOR=`(eval "$ac_cpp -traditional-cpp conftest.$ac_ext") 2>&AS_MESSAGE_LOG_FD | $EGREP "^version" | sed "s,version\ ,,"`
 	rm -f conftest*
 
 	AC_MSG_NOTICE([found dvb version $DVB_API_VERSION.$DVB_API_VERSION_MINOR])
-- 
2.1.4

