From 74897e01673de934b19934ed770dacefa08c88af Mon Sep 17 00:00:00 2001
From: MarkusVolk <f_l_k@t-online.de>
Date: Thu, 15 Oct 2015 18:42:07 +0200
Subject: [PATCH 1424/1424] fix build for new ffmpeg

Signed-off-by: MarkusVolk <f_l_k@t-online.de>
---
 src/driver/record.cpp | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/src/driver/record.cpp b/src/driver/record.cpp
index c91f67a..99850dc 100644
--- a/src/driver/record.cpp
+++ b/src/driver/record.cpp
@@ -64,6 +64,19 @@
 #include <timerdclient/timerdclient.h>
 #include <cs_api.h>
 
+#if LIBAVCODEC_VERSION_INT <= AV_VERSION_INT(57, 1, 99)
+	#define CODEC_ID_AC3  CODEC_ID_AC3
+	#define CODEC_ID_AAC  CODEC_ID_AAC
+	#define CODEC_ID_EAC3 CODEC_ID_EAC3
+	#define CODEC_ID_MP2  CODEC_ID_MP2
+	#define CODEC_ID_H264 CODEC_ID_H264
+#else
+	#define CODEC_ID_AC3  AV_CODEC_ID_AC3
+	#define CODEC_ID_AAC  AV_CODEC_ID_AAC
+	#define CODEC_ID_EAC3 AV_CODEC_ID_EAC3
+	#define CODEC_ID_MP2  AV_CODEC_ID_MP2
+	#define CODEC_ID_H264 AV_CODEC_ID_H264
+#endif
 /* TODO:
  * nextRecording / pending recordings - needs testing
  * check/fix askUserOnTimerConflict gui/timerlist.cpp -> getOverlappingTimers lib/timerdclient/timerdclient.cpp
-- 
2.1.4

