From 4cea1d8dcba013f83fe3dbd1a35f326a911be890 Mon Sep 17 00:00:00 2001
From: MarkusVolk <f_l_k@t-online.de>
Date: Sun, 20 Dec 2015 09:49:39 +0100
Subject: [PATCH 82/82] sectionsd: remove #ifdef, unneeded for uclibc-ng

Signed-off-by: MarkusVolk <f_l_k@t-online.de>
---
 src/eitd/sectionsd.cpp | 14 +-------------
 1 file changed, 1 insertion(+), 13 deletions(-)

diff --git a/src/eitd/sectionsd.cpp b/src/eitd/sectionsd.cpp
index 3df0860..8207417 100644
--- a/src/eitd/sectionsd.cpp
+++ b/src/eitd/sectionsd.cpp
@@ -1065,11 +1065,7 @@ static void commandDumpStatusInformation(int /*connfd*/, char* /*data*/, const u
 		 //    resourceUsage.ru_maxrss, resourceUsage.ru_ixrss, resourceUsage.ru_idrss, resourceUsage.ru_isrss,
 		);
 	printf("%s\n", stati);
-#ifdef __UCLIBC__
-	malloc_stats(NULL);
-#else
 	malloc_stats();
-#endif
 	return ;
 }
 
@@ -1195,12 +1191,8 @@ static void FreeMemory()
 	mySIeventsNVODorderUniqueKey.clear();
 
 	unlockEvents();
-
-#ifdef __UCLIBC__
-	malloc_stats(NULL);
-#else
 	malloc_stats();
-#endif
+
 	xprintf("[sectionsd] free memory done\n");
 	//wakeupAll(); //FIXME should we re-start eit here ?
 }
@@ -2050,11 +2042,7 @@ static void print_meminfo(void)
 	if (!sections_debug)
 		return;
 
-#ifdef __UCLIBC__
-	malloc_stats(NULL);
-#else
 	malloc_stats();
-#endif
 }
 
 //---------------------------------------------------------------------
-- 
2.1.4

