From a5edd55cb0a2591e6fb616a0c009e32beffd4750 Mon Sep 17 00:00:00 2001
From: bazi98 <bazi98@web.de>
Date: Thu, 22 Oct 2015 08:22:42 +0200
Subject: [PATCH 4/4] update olive.theme

---
 data/themes/Olive.theme | 31 +++++++++++++++++++++++++++----
 1 file changed, 27 insertions(+), 4 deletions(-)

diff --git a/data/themes/Olive.theme b/data/themes/Olive.theme
index 9d7f7f3..36270e4 100644
--- a/data/themes/Olive.theme
+++ b/data/themes/Olive.theme
@@ -1,11 +1,33 @@
+clock_Digit_alpha=0
+clock_Digit_blue=100
+clock_Digit_green=100
+clock_Digit_red=100
+colored_events_alpha=0
+colored_events_blue=0
+colored_events_channellist=0
+colored_events_green=70
+colored_events_infobar=0
+colored_events_red=95
 infobar_Text_alpha=0
 infobar_Text_blue=20
 infobar_Text_green=20
 infobar_Text_red=20
 infobar_alpha=0
 infobar_blue=80
+infobar_casystem_alpha=8
+infobar_casystem_blue=0
+infobar_casystem_green=0
+infobar_casystem_red=0
+infobar_gradient_body=1
+infobar_gradient_body_direction=1
+infobar_gradient_bottom=3
+infobar_gradient_bottom_direction=1
+infobar_gradient_top=1
+infobar_gradient_top_direction=1
 infobar_green=80
 infobar_red=80
+menu_ButtonBar_gradient=0
+menu_ButtonBar_gradient_direction=1
 menu_Content_Selected_Text_alpha=0
 menu_Content_Selected_Text_blue=0
 menu_Content_Selected_Text_green=0
@@ -36,9 +58,10 @@ menu_Head_Text_green=20
 menu_Head_Text_red=20
 menu_Head_alpha=0
 menu_Head_blue=80
+menu_Head_gradient=2
+menu_Head_gradient_direction=1
 menu_Head_green=80
 menu_Head_red=80
-menu_Hint_gradient=0
-menu_Head_gradient=0
-infobar_gradient_top=0
-infobar_gradient_bottom=0
+menu_Hint_gradient=3
+menu_Hint_gradient_direction=0
+menu_Separator_gradient_enable=1
-- 
2.6.2

