From dd8d7c5a36cc1783892e97e39b3259cecdb5024f Mon Sep 17 00:00:00 2001
From: Thilo Graf <dbt@novatux.de>
Date: Wed, 18 Nov 2015 19:25:36 +0100
Subject: [PATCH 09/21] CTimeOSD: disable setHeight() in CTimeOSD, height is 
 bound to settings

---
 src/gui/timeosd.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/gui/timeosd.h b/src/gui/timeosd.h
index b46fdbe..cf15b76 100644
--- a/src/gui/timeosd.h
+++ b/src/gui/timeosd.h
@@ -60,5 +60,6 @@ class CTimeOSD : public CComponentsFrmClock
 		void switchMode(int position, int duration);
 		mode getMode() { return m_mode; };
 		void setMode (mode mode_) { m_mode = mode_; };
+		void setHeight(const int){}//NOTE: dummy member, height is strictly bound to settings
 };
 #endif
-- 
2.1.4

