From 9937b75a0b4b77422f4c1e1f692b781abd1b4591 Mon Sep 17 00:00:00 2001
From: MarkusVolk <f_l_k@t-online.de>
Date: Tue, 2 Feb 2016 15:44:14 +0100
Subject: [PATCH] fix

Signed-off-by: MarkusVolk <f_l_k@t-online.de>
---
 src/gui/opkg_manager.cpp | 21 ++++++++++-----------
 1 file changed, 10 insertions(+), 11 deletions(-)

diff --git a/src/gui/opkg_manager.cpp b/src/gui/opkg_manager.cpp
index 46c1ad0..7e0d657 100644
--- a/src/gui/opkg_manager.cpp
+++ b/src/gui/opkg_manager.cpp
@@ -58,18 +58,13 @@
 #include <sys/wait.h>
 #include <fstream>
 
-#if 0
 #define OPKG_CL "opkg"
-#else
-#define OPKG_CL "opkg-cl"
-#endif
-
 #define OPKG_TMP_DIR "/tmp/.opkg"
 #define OPKG_TEST_DIR OPKG_TMP_DIR "/test"
 #define OPKG_CL_CONFIG_OPTIONS " -V2 --tmp-dir=/tmp --cache=" OPKG_TMP_DIR
 
 #define OPKG_BAD_PATTERN_LIST_FILE CONFIGDIR "/bad_package_pattern.list"
-#define OPKG_CONFIG_FILE "/etc/opkg/opkg.conf"
+#define OPKG_CONFIG_FILE "/etc/opkg/opkg.conf.broken"
 
 using namespace std;
 
@@ -95,10 +90,10 @@ static const string pkg_types[OM_MAX] =
 	OPKG_CL " list-installed ",
 	OPKG_CL " list-upgradable ",
 	OPKG_CL " -A update ",
-	OPKG_CL OPKG_CL_CONFIG_OPTIONS " upgrade ",
-	OPKG_CL OPKG_CL_CONFIG_OPTIONS " remove ",
+	OPKG_CL OPKG_CL_CONFIG_OPTIONS " --tmp-dir=/tmp --cache=/tmp --force-depends upgrade ",
+	OPKG_CL OPKG_CL_CONFIG_OPTIONS " --tmp-dir=/tmp --cache=/tmp --force-depends  remove ",
 	OPKG_CL " info ",
-	OPKG_CL OPKG_CL_CONFIG_OPTIONS " install ",
+	OPKG_CL OPKG_CL_CONFIG_OPTIONS "  --tmp-dir=/tmp --cache=/tmp  --force-depends --force-reinstall install ",
 	OPKG_CL " status ",
 	OPKG_CL " configure ",
 	OPKG_CL " download "
@@ -526,11 +521,11 @@ int COPKGManager::showMenu()
 	menu->addItem(fw);
 
 	//feed setup
-	CMenuWidget feeds_menu(LOCALE_OPKG_TITLE, NEUTRINO_ICON_UPDATE, w_max (100, 10));
+	/*CMenuWidget feeds_menu(LOCALE_OPKG_TITLE, NEUTRINO_ICON_UPDATE, w_max (100, 10));
 	showMenuConfigFeed(&feeds_menu);
 	fw = new CMenuForwarder(LOCALE_OPKG_FEED_ADDRESSES, true, NULL, &feeds_menu, NULL, CRCInput::RC_www);
 	fw->setHint(NEUTRINO_ICON_HINT_SW_UPDATE, LOCALE_MENU_HINT_OPKG_FEED_ADDRESSES_EDIT);
-	menu->addItem(fw);
+	menu->addItem(fw);*/
 
 	menu->addItem(GenericMenuSeparatorLine);
 
-- 
2.1.4

