From 5ebea66cdfaeab45915d6971d7cdc29d039f75f3 Mon Sep 17 00:00:00 2001
From: flk <f_l_k@t-online.de>
Date: Tue, 16 Feb 2016 16:15:41 +0100
Subject: [PATCH 1/2] opkg-0.3.x-uses-opkg-instead-of-opkg-cl-as-binary-name

Signed-off-by: flk <f_l_k@t-online.de>
---
 src/gui/opkg_manager.cpp | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/gui/opkg_manager.cpp b/src/gui/opkg_manager.cpp
index 4afbe1d..889fd4d 100644
--- a/src/gui/opkg_manager.cpp
+++ b/src/gui/opkg_manager.cpp
@@ -58,18 +58,14 @@
 #include <sys/wait.h>
 #include <fstream>
 
-#if 0
 #define OPKG_CL "opkg"
-#else
-#define OPKG_CL "opkg-cl"
-#endif
 
 #define OPKG_TMP_DIR "/tmp/.opkg"
 #define OPKG_TEST_DIR OPKG_TMP_DIR "/test"
 #define OPKG_CL_CONFIG_OPTIONS " -V2 --tmp-dir=/tmp --cache=" OPKG_TMP_DIR
 
 #define OPKG_BAD_PATTERN_LIST_FILE CONFIGDIR "/bad_package_pattern.list"
-#define OPKG_CONFIG_FILE "/etc/opkg/opkg.conf"
+#define OPKG_CONFIG_FILE "/etc/opkg/opkg.conf.broken"
 
 using namespace std;
 
@@ -91,10 +91,10 @@ static const string pkg_types[OM_MAX] =
 	OPKG_CL " list-installed ",
 	OPKG_CL " list-upgradable ",
 	OPKG_CL " -A update ",
-	OPKG_CL OPKG_CL_CONFIG_OPTIONS " upgrade ",
-	OPKG_CL OPKG_CL_CONFIG_OPTIONS " remove ",
+	OPKG_CL OPKG_CL_CONFIG_OPTIONS " --tmp-dir=/tmp --cache=/tmp --force-depends upgrade ",
+	OPKG_CL OPKG_CL_CONFIG_OPTIONS " --tmp-dir=/tmp --cache=/tmp --force-depends  remove ",
 	OPKG_CL " info ",
-	OPKG_CL OPKG_CL_CONFIG_OPTIONS " install ",
+	OPKG_CL OPKG_CL_CONFIG_OPTIONS "  --tmp-dir=/tmp --cache=/tmp  --force-depends --force-reinstall install ",
 	OPKG_CL " status ",
 	OPKG_CL " configure ",
 	OPKG_CL " download "
-- 
2.1.4


