From 609a591d9c06bfe287b30046acbc551fbae4ad75 Mon Sep 17 00:00:00 2001
From: Stefan Seyfried <seife@tuxbox-git.slipkontur.de>
Date: Fri, 1 Jan 2016 19:30:58 +0100
Subject: [PATCH 7995/8016] opkg_manager: disarm checkSize() function

Signed-off-by: MarkusVolk <f_l_k@t-online.de>
---
 src/gui/opkg_manager.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/gui/opkg_manager.cpp b/src/gui/opkg_manager.cpp
index 56933ea..06bccf6 100644
--- a/src/gui/opkg_manager.cpp
+++ b/src/gui/opkg_manager.cpp
@@ -249,6 +249,13 @@ bool COPKGManager::checkSize(const string& pkg_name)
 	if (isInstalled(plain_pkg))
 		return true;
 
+	/* this is pretty broken right now for several reasons:
+	   * space in /tmp is limited (/tmp being ramfs usually, but wasted
+	     by unpacking the archive and then untaring it instead of using a pipe
+	   * the file is downloaded for this test, then discarded and later
+	     downloaded again for installation
+	   so until a better solution is found, simply disable it.  */
+#if 0
 	//get available root fs size
 	//TODO: Check writability!
 	struct statfs root_fs;
@@ -303,6 +310,7 @@ bool COPKGManager::checkSize(const string& pkg_name)
 		dprintf(DEBUG_NORMAL,  "[COPKGManager] [%s - %d]  WARNING: size check freesize=%" PRId64 " (recommended: %" PRId64 ")\n", __func__, __LINE__, free_size, req_size);
 		return false;
 	}
+#endif
 	return true;
 }
 
-- 
2.1.4

