From 664e3012b310d38e81efe225b7c5e61d4bd92af2 Mon Sep 17 00:00:00 2001
From: Stefan Seyfried <seife@tuxbox-git.slipkontur.de>
Date: Fri, 1 Jan 2016 23:18:59 +0100
Subject: [PATCH 8003/8016] opkg_manager: explicitly clear pkg_map in
 destructor

Signed-off-by: MarkusVolk <f_l_k@t-online.de>
---
 src/gui/opkg_manager.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/gui/opkg_manager.cpp b/src/gui/opkg_manager.cpp
index 2597dc7..79b951d 100644
--- a/src/gui/opkg_manager.cpp
+++ b/src/gui/opkg_manager.cpp
@@ -137,6 +137,7 @@ void COPKGManager::init()
 
 COPKGManager::~COPKGManager()
 {
+	pkg_map.clear();
 	CFileHelpers::removeDir(OPKG_TMP_DIR);
 }
 
-- 
2.1.4

